modules/dnn/src/layers/batch_norm_layer.cpp

```diff
@@ -150,9 +153,9 @@ class BatchNormLayerImpl CV_FINAL : public BatchNormLayer
 
     virtual bool supportBackend(int backendId) CV_OVERRIDE
     {
-        return backendId == DNN_BACKEND_OPENCV ||
+        return (backendId == DNN_BACKEND_OPENCV && (dims == 4 || dims == 2)) ||
```

Comment:
Have you tried to run BN with 5D input? Looks a bit confusing for me why it works with 4 and 2 dimensions but does not work with 3 and 5.